go/service/user.go

```diff
@@ -344,6 +345,179 @@ func (h *UserHandler) UploadUserAvatar(ctx context.Context, arg keybase1.UploadU
+		if serviceType == nil {
+			mctx.CDebugf("missing proof service type: %v", service)
+			continue
+		}
+		if len(user.IDTable().GetActiveProofsFor(serviceType)) > 0 {
+			mctx.CDebugf("user has an active proof: %v", serviceType.Key())
+			continue
+		}
+		subtext := serviceType.DisplayGroup()
+		if len(subtext) == 0 {
+			subtext = serviceType.PickerSubtext()
+		}
+		suggestions = append(suggestions, ProofSuggestion{ProofSuggestion: keybase1.ProofSuggestion{
+			Key:           service,
+			ProfileText:   fmt.Sprintf("Prove your %v", serviceType.DisplayName()),
+			ProfileIcon:   dummyIcon,
+			PickerText:    serviceType.DisplayName(),
+			PickerSubtext: subtext,
+			PickerIcon:    dummyIcon,
+		}})
+	}
+	hasPGP := len(user.GetActivePGPKeys(true)) > 0
+	if !hasPGP {
+		suggestions = append(suggestions, ProofSuggestion{ProofSuggestion: keybase1.ProofSuggestion{
+			Key:           "pgp",
+			ProfileText:   "Add a PGP key",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "PGP key",
+			PickerSubtext: "",
+			PickerIcon:    dummyIcon,
+		}})
+	}
+	// Always show the option to create a new web proof.
+	suggestions = append(suggestions, ProofSuggestion{ProofSuggestion: keybase1.ProofSuggestion{
+		Key:           "web",
+		ProfileText:   "Prove your website",
+		ProfileIcon:   dummyIcon,
+		PickerText:    "Your own website",
+		PickerSubtext: "",
+		PickerIcon:    dummyIcon,
+	}})
+	if !user.IDTable().HasActiveCryptocurrencyFamily(libkb.CryptocurrencyFamilyBitcoin) {
+		suggestions = append(suggestions, ProofSuggestion{ProofSuggestion: keybase1.ProofSuggestion{
+			Key:           "bitcoin",
+			ProfileText:   "Set a Bitcoin address",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Bitcoin address",
+			PickerSubtext: "",
+			PickerIcon:    dummyIcon,
+		}})
+	}
+	if !user.IDTable().HasActiveCryptocurrencyFamily(libkb.CryptocurrencyFamilyZCash) {
+		suggestions = append(suggestions, ProofSuggestion{ProofSuggestion: keybase1.ProofSuggestion{
+			Key:           "zcash",
+			ProfileText:   "Set a Zcash address",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Zcash address",
+			PickerSubtext: "",
+			PickerIcon:    dummyIcon,
+		}})
+	}
+
+	// Alphabetize so that ties later on in SliceStable are deterministic.
+	sort.Slice(suggestions, func(i, j int) bool {
+		return suggestions[i].Key < suggestions[j].Key
+	})
+
+	// Priorities from the server.
+	serverPriority := make(map[string]int) // key -> server priority
+	maxServerPriority := 0
+	for _, displayConfig := range mctx.G().GetProofServices().ListDisplayConfigs() {
+		if displayConfig.Priority <= 0 {
+			continue
+		}
+		var altKey string
+		switch displayConfig.Key {
+		case "zcash.t", "zcash.z", "zcash.s":
+			altKey = "zcash"
+		case "http", "https", "dns":
+			altKey = "web"
+		}
+		serverPriority[displayConfig.Key] = displayConfig.Priority
+		if len(altKey) > 0 {
+			if _, ok := serverPriority[altKey]; !ok {
+				serverPriority[altKey] = displayConfig.Priority
```

Comment:
should this be the min/max value of the `altKey` instead of just the first instance of it?